sprouts.py

```diff
@@ -0,0 +1,246 @@
+                            
+                            else:
+                                DrawingLine = True
+                                current_line = []
+                                current_line.append(start_line)
+                                #Updates text box:
+                        
+                        #Ends line
+                        elif DrawingLine == True: 
+                            DrawingLine = False
+                            LineMode = False
+                            DotMode = True
+                            redraw_box()
+                            text_writing('Place dot on the line just drawn', ss.ORANGE, 500, 25, 25)
+                    
+                    #Draws dot
+                    elif DotMode: 
+                        currentdotposition = event.pos
+                        if check_dots(currentdotposition) == False: #Function: if placed on existing dot
+                            #Updates text box:
+                            error_display(True, False)
+                        
+                        else:
+                            #Draws dot:
+                            DrawingLine = False
+                            if distance_check == True: #<--- This function creates problem.
+                                draw_dot(currentdotposition, ss.BLUE)
+                                dot_pos.append(currentdotposition) #Saves location
+                                redraw_box()
+                                text_writing('Draw a line starting from any dot', ss.ORANGE, 500, 25, 25)
+
+                            else:
+                                #Updates text box: (Will implement in error function later)
+                                redraw_box()
+                                text_writing('Error: Place dot on line', ss.ORANGE, 500, 25, 25)
+                                
+
+                            #Switches back to line drawing:
+                            DotMode = False
+                            LineMode = True
+                            
+
+            #Connects start and end with line:
+            elif event.type == pygame.MOUSEMOTION: #Move mouse
+                if DrawingLine:
+                    current_point = pygame.mouse.get_pos()
+                    if distance(start_line, current_point) > 30: #Only starts drawing if 30pix away from dot, resulting in less coordinates
+                        pygame.draw.line(win, ss.BLACK, start_line, current_point)
+                        start_line = current_point 
+                        current_line.append(start_line) #Saves line location
+                    
+                 
+        pygame.display.update()
+    
+        clock.tick(10) #10 Frames per second
+
+
+####FUNCTIONS
+
+#Writes all text 
+def text_writing(message,colour, x, y, size):
+    font = pygame.font.SysFont('Arial.ttf', size)
+    text = font.render(message, False, colour)
```

Comment:
If you set this to True the text will be smoother (less pixelated)

Reply:
Great tip!